.github/workflows/build-test.yml

```
@@ -0,0 +1,176 @@
+name: Build Test
+
+on:
+  pull_request:
+    paths-ignore:
+    - README.md
+    - VERSION_INFO
+    - .ci/*
+    - .readthedocs.yml
+    - docs-src/*
+    - docs-img/*
+    - docs-jupyter/*
+    - docs-doxygen/*
+    - docs-sphinx/*
+    - docs/*
+    - studies/*
+    - src/cuda-kernels
+    - cuda-kernels
+  # Don't run for any PR
+  schedule:
+  - cron: '0 12 1 * *'
+
+  workflow_dispatch:
+
+concurrency:
+  group: build-test-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  Windows:
+    strategy:
+      matrix:
+        python-version:
+        - '3.9'
+        - '3.10'
+        - '3.9'
+        - '3.8'
+        - '3.7'
+        - '3.6'
+    runs-on: windows-2019
+    env:
+      PIP_ONLY_BINARY: cmake
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+    - name: Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install Pip
+      run: python -m pip install --upgrade pip
+    - name: Print versions
+      run: python -m pip list
+    - name: Build
+      run: python -m pip install -v .[test,dev]
+    - name: Check if kernel specification is sorted
+      run: python dev/kernel-diagnostics.py --check-spec-sorted
+    - name: Generate Kernel tests
+      run: python dev/generate-tests.py
+    - name: Test specification
+      run: python -m pytest -vv -rs tests-spec
+    - name: Test specification with explicitly defined values
+      run: python -m pytest -vv -rs tests-spec-explicit
+    - name: Test CPU kernels
+      run: python -m pytest -vv -rs tests-cpu-kernels
+    - name: Test
+      run: python -m pytest -vv -rs tests
+
+  MacOS:
+    strategy:
+      matrix:
+        python-version:
+        - '3.10'
+        - '3.9'
+        - '3.8'
+        - '3.7'
+        - '3.6'
+    runs-on: macOS-10.15

```

```suggestion
    runs-on: macOS-11
```

10.15 is deprecated.